File: src/Python/374.GuessNumber.py

def guess(num):
    NUM = 6
    if num == NUM:
        return 0
    if num < NUM:
        return 1
    if num > NUM:
        return -1

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left = 1
        right = n
        while left <= right:
            middle = left + (right - left) // 2
            num = middle
            if guess(num) == 0:
                return num
            if guess(num) == 1:
                left = middle + 1
            if guess(num) == -1:
                right = middle - 1
        return left
            
if __name__ == '__main__':
    solution = Solution()
    print(solution.guessNumber(10))
else:
    pass

File: src/CSharp/374.GuessNumber.cs
using System;
using System.Collections.Generic;

namespace PeefyLeetCode.GuessNumber
{
    public abstract class GuessGame
    {
        public virtual int Guess(int num);
    }
public class Solution : GuessGame {
    public int GuessNumber(int n) {
        int left = 1;
        int right = n;
        while (left <= right){
            int middle = left + (right - left) / 2;
            int num = middle;
            if (Guess(num) == 0)
                return num;
            if (Guess(num) == 1)
                left = middle + 1;
            if (Guess(num) == -1)
                right = middle - 1;
        }
        return left;
    }
}
}